File: app/models/table.py
import uuid
from typing import TYPE_CHECKING

from sqlalchemy import Boolean, ForeignKey, Integer, String, UniqueConstraint, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base

if TYPE_CHECKING:
    from app.models.branch import Branch


class Table(Base):
    __tablename__ = "tables"
    __table_args__ = (UniqueConstraint("branch_id", "code"),)

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()")
    )
    branch_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("branches.id", ondelete="CASCADE"), nullable=False
    )
    code: Mapped[str] = mapped_column(String(20), nullable=False)
    capacity: Mapped[int] = mapped_column(Integer, nullable=False, server_default="4")
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default=text("true"))

    branch: Mapped["Branch"] = relationship()

File: php/src/Models/Table.php
<?php

declare(strict_types=1);

namespace App\Models;

use App\Core\Row;

final class Table
{
    public function __construct(
        public readonly string $id,
        public readonly string $branchId,
        public readonly string $code,
        public readonly int $capacity,
        public readonly bool $isActive,
    ) {
    }

    /** @param array<string, mixed> $row */
    public static function fromRow(array $row): self
    {
        return new self(
            id: $row['id'],
            branchId: $row['branch_id'],
            code: $row['code'],
            capacity: (int) $row['capacity'],
            isActive: Row::bool($row['is_active']),
        );
    }
}
